library/wizard/card_no.py

```diff
@@ -0,0 +1,33 @@
+# See LICENSE file for full copyright and licensing details.
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class CardNumber(models.TransientModel):
+    """Defining Card Number."""
+
+    _name = "card.number"
+    _description = "Card Number"
+
+    card_id = fields.Many2one(
+        "library.card", "Card No", required=True, help="Library card"
+    )
+
+    def card_number_ok(self):
+        lib_book_obj = self.env["library.book.issue"]
+        for rec in self:
+            search_card_rec = lib_book_obj.search(
```

Comment:
we don't need variable "**search_card_rec**" here. Below code will work.
if not lib_book_obj.search(
                [("card_id", "=", rec.card_id.id)]
            ):
            raise UserError(_("Invalid Card Number!"))
else:
    return {
                    "type": "ir.actions.act_window",
                    "res_model": "book.name",
                    "src_model": "library.book.issue",
                    "target": "new",
                    "view_mode": "form",
                    "context": {"default_card_id": rec.card_id.id},
                }